samples/valueentity-counter-with-action-controller/src/main/java/com/example/actions/DoubleCounterAction.java

```diff
@@ -0,0 +1,41 @@
+/* This code was generated by Akka Serverless tooling.
+ * As long as this file exists it will not be re-generated.
+ * You are free to make changes to this file.
+ */
+
+package com.example.actions;
+
+import com.akkaserverless.javasdk.ServiceCallRef;
+import com.akkaserverless.javasdk.action.ActionCreationContext;
+import com.example.CounterApi;
+import com.google.protobuf.Empty;
+
+// tag::double-counter-action[]
+/** An action. */
+public class DoubleCounterAction extends AbstractDoubleCounterAction {
+
+  private final ActionCreationContext creationContext; // <1>
+
+  public DoubleCounterAction(ActionCreationContext creationContext) {
+    this.creationContext = creationContext;
+  }
+
+  /** Handler for "Increase". */
+  @Override
+  public Effect<Empty> increase(CounterApi.IncreaseValue increaseValue) {
+
+    int doubled = increaseValue.getValue() * 2;
+    CounterApi.IncreaseValue increaseValueDoubled =
+        increaseValue.toBuilder().setValue(doubled).build(); // <2>
+
+    ServiceCallRef<CounterApi.IncreaseValue> callRef = // <3>
+        creationContext.serviceCallFactory()
+          .lookup(
+              "com.example.CounterService",
+              "Increase",
+              CounterApi.IncreaseValue.class);
```

Comment:
can this be done in the constructor instead? for actions it is no runtime advantage since we create a new instance for each request, but isn't it conceptually right to initialize the ServiceCallRef once and then use it for each call?

Reply:
I had this in mind when I started, but I changed my mind when I realised that the lookup is per method, not per service. 

I don't know if it's conceptually right. It felt to me easier to put it in the method so we don't need to explain why we are creating it in the constructor. Anyway, I don't mind much. Hopefully this goes away soon.

Reply:
> since we create a new instance for each request

That's actually our solution for the `ActionContext` and the async boundaries issue https://github.com/lightbend/akkaserverless-java-sdk/issues/268

We should have only one context passed to the constructor.